app/routes/questionnaire.py

```diff
@@ -165,23 +169,27 @@ def post_questionnaire(schema, questionnaire_store):
 @with_questionnaire_store
 @with_schema
 def get_section(schema, questionnaire_store, section_id, list_item_id=None):
-    progress_store = questionnaire_store.progress_store
     router = Router(
-        schema, progress_store=progress_store, list_store=questionnaire_store.list_store
+        schema,
+        questionnaire_store.answer_store,
+        questionnaire_store.list_store,
+        questionnaire_store.progress_store,
+        questionnaire_store.metadata,
     )
 
     if not schema.is_hub_enabled():
         redirect_location = router.get_first_incomplete_location_in_survey()
         return redirect(redirect_location.url())
+
     section = schema.get_section(section_id)
 
-    if not section:
+    if not section or section_id not in router.enabled_section_ids:
         return redirect(url_for('.get_questionnaire'))
 
-    routing_path = path_finder.routing_path(
+    routing_path = router.path_finder.routing_path(
```

Comment:
We should consider hiding path_finder from callers of router. We could add a `section_routing_path` method on router that calls `path_finder.routing_path`.

Reply:
Have made it a method in router as I think it looks cleaner. So right now, only router and calculated summary used `PathFinder`, but could look at dropping `PathFinder` from calculated summary in favour of `Router` if we think `PathFinder` must be used via router?